### PR TITLE
ci: Move file change filter definitions to a file

### PR DESCRIPTION
**Pull request by aborgna-q**
Cleanup PR. Centralises the file filter used by `dorny/paths-filter` when detecting changes in a single `.yml`.

### CODE REVIEW

File: .github/workflows/ci-py.yml [hunk 1]
```diff
@@ -31,11 +31,7 @@ jobs:
     - uses: dorny/paths-filter@v3
       id: filter
       with:
-        filters: |
-          python:
-            - 'hugr-py/**'
-            - 'pyproject.toml'
-            - 'specification/schema/**'
+        filters: .github/change-filters.yml
```
Comment (croyzor) [inline .github/workflows/ci-py.yml:34]: Do we not need to specify whether we want the rust filter or the python filter here?

Comment (aborgna-q) [reply .github/workflows/ci-py.yml:34]: It computes both, an we use the one we need. It's not an expensive operation so it doesn't change much.

Comment (croyzor) [reply .github/workflows/ci-py.yml:34]: So after this PR, we'll run both `ci-py` and `ci-rs` when relevant python or rust files are changed, but we wont run either when other files change. Is that right?

Comment (aborgna-q) [reply .github/workflows/ci-py.yml:34]: Yes, that's the same behaviour we have already.

The job steps check for
```yaml
if: ${{ needs.changes.outputs.python == 'true' }}
```
or
```yaml
if: ${{ needs.changes.outputs.rust == 'true' }}
```
to see if they should run or be skipped

**Review by croyzor — APPROVED** [review]

**Pull request merged**